.github/workflows/pragmaonce.yml

```diff
@@ -0,0 +1,30 @@
+name: pragma-once checker
+
+on: [pull_request_target]
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        ref: ${{ github.event.pull_request.head.sha }}
+        persist-credentials: false
+        # We need the history of the dev branch all the way back to where the PR
+        # diverged. We're fetching everything here, as we don't know how many
+        # commits back that point is.
+        fetch-depth: 0
```

Comment:
We're only checking the latest commit, so we don't need to fetch everything. You should be able to remove the `fetch-depth` option for a speedup.